haskell/cabal
# Guard PackageInfo behind `cabal-version` ≥ 3.12

Towards #9331.
I took the liberty to to run with @andreabedini #9374 as the Rubenesque #8427 got merged meanwhile, meaning the former PR needed to be updated.

Once we are fine with this patch, there is the question of whether — and how — to backport it to 3.10.

Include the following checklist in your PR:


---

**QA notes**

- create a new folder, cd into it (e.g. `mkdir prova && cd prova`)
- `cabal init -nm` and then modify the file to have `PackageInfo_pkg` modules. Example follows:

        cabal-version: 2.4
        name: prova
        version: 0
        license: GPL-3.0-or-later
        maintainer: Someone
        category: Example
        synopsis: Foo
        description: FooBar
        build-type: Simple
        extra-doc-files: CHANGELOG.md

        library
            default-language: Haskell2010
            build-depends: base <5
            -- ☞ N.B.: PackageInfo packages must contain the same name of
            -- of the package! (In this example: `prova`).
            autogen-modules: PackageInfo_prova
            exposed-modules: PackageInfo_prova
- `cabal` v3.10.*: run `cabal check`, no error/warnings about PackageInfo modules is reported
- `cabal` master: run `cabal check`, you will be warned about having to use `cabal-version` 3.12 or higher
